src/presentation/pages/Join.vue

```diff
@@ -0,0 +1,19 @@
+<template>
+  <div>
+    <h1>Join</h1>
+  </div>
+</template>
+
+
+<script lang="ts" setup>
+import { onMounted } from 'vue';
+import { useRoute } from 'vue-router';
+import useNoteSettings from '@/application/services/useNoteSettings';
+
+const route = useRoute();
+const { joinNote } = useNoteSettings();
+
+onMounted(async () => {
```

Comment:
maybe onBeforeRouteEnter?

Reply:
yes, but onBeforeRouteEnter is not available on composition api but on option api

except i need to import application service into routes.ts file

Reply:
Read this guide, it can help
https://router.vuejs.org/guide/advanced/navigation-guards.html#The-Full-Navigation-Resolution-Flow

Probably, `beforeEnter` can be used here

Reply:
its can but in routes.ts file

Reply:
this comment is still actual. Please, read the whole article from Vue Router doc and select a corresponded hook